xmtp_mls/src/groups/commit_log.rs

```diff
@@ -344,46 +316,248 @@ where
+        conversation_id: &[u8],
+    ) -> Result<ForkedStateCheckResult, CommitLogError> {
+        // Get cursors for this conversation
+        let fork_check_local_cursor = conn.get_last_cursor_for_id(
+            conversation_id,
+            xmtp_db::refresh_state::EntityKind::CommitLogForkCheckLocal,
+        )?;
+        let fork_check_remote_cursor = conn.get_last_cursor_for_id(
+            conversation_id,
+            xmtp_db::refresh_state::EntityKind::CommitLogForkCheckRemote,
+        )?;
+
+        // Get local and remote commit logs
+        let local_logs = conn.get_local_commit_log_after_cursor(
+            conversation_id,
+            fork_check_local_cursor,
+            LocalCommitLogOrder::DescendingByRowid,
+        )?;
+        let remote_logs =
+            conn.get_remote_commit_log_after_cursor(conversation_id, fork_check_remote_cursor)?;
+
+        // Check each local log against remote logs for matching commit_sequence_id
+        for local_log in &local_logs {
+            if let Some(matching_remote_log) =
+                self.find_matching_remote_log(&remote_logs, local_log.commit_sequence_id)
+            {
+                // Found a matching commit_sequence_id - check if forked
+                let is_forked = local_log.applied_epoch_authenticator
+                    != matching_remote_log.applied_epoch_authenticator;
+
+                if is_forked {
+                    println!("Forked state check result: {:?}", matching_remote_log);
+                    println!("Local log: {:?}", local_log);
+                }
+
+                // Update cursors regardless of fork status (we found a match)
+                conn.update_cursor(
+                    conversation_id,
+                    xmtp_db::refresh_state::EntityKind::CommitLogForkCheckLocal,
+                    local_log.rowid as i64,
+                )?;
+                conn.update_cursor(
+                    conversation_id,
+                    xmtp_db::refresh_state::EntityKind::CommitLogForkCheckRemote,
+                    matching_remote_log.rowid as i64,
+                )?;
+
+                // Return the result
+                return Ok(ForkedStateCheckResult {
+                    is_forked,
+                    forked_epoch_number: if is_forked {
+                        Some(matching_remote_log.applied_epoch_number as u64)
+                    } else {
+                        None
+                    },
+                    forked_commit_sequence_id: if is_forked {
+                        Some(matching_remote_log.commit_sequence_id as u64)
+                    } else {
+                        None
+                    },
+                });
+            }
+        }
+
+        // No matching commit_sequence_id found for any local log = no fork detected
+        Ok(ForkedStateCheckResult {
+            is_forked: false,
+            forked_epoch_number: None,
+            forked_commit_sequence_id: None,
+        })
```

Comment:
You're probably already planning to do this, but once we persist the fork state, once a group is forked we probably shouldn't run this fn again, otherwise we might accidentally return `is_forked: false` here if there are no new entries